src/models/split_transfers.py

```diff
@@ -99,39 +115,44 @@ def _process_token_transfers(
                     # Reinsert since there is still an amount owed.
                     self.overdrafts[solver] = overdraft
                     continue
-            if solver in cow_redirects:
-                # Redirect COW rewards to reward target specific by VouchRegistry
-                redirect_address = cow_redirects[solver].reward_target
-                log_saver.print(
-                    f"Redirecting solver {solver} COW tokens "
-                    f"({transfer.amount}) to {redirect_address}",
-                    category=Category.REDIRECT,
-                )
-                transfer.receiver = redirect_address
+            transfer.redirect_to(redirect_map, self.log_saver)
             self.cow_transfers.append(transfer)
+        # We do not need to worry about any controversy between overdraft
+        # and positive slippage adjustments, because positive/negative slippage
+        # is disjoint between solvers.
+        while positive_slippage:
+            slippage_transfer = Transfer.from_slippage(positive_slippage.pop())
+            slippage_transfer.redirect_to(redirect_map, self.log_saver)
+            self.eth_transfers.append(slippage_transfer)
 
     def process(
         self,
-        indexed_slippage: dict[Address, SolverSlippage],
+        slippage: SplitSlippages,
         cow_redirects: dict[Address, Vouch],
-        log_saver: PrintStore,
     ) -> list[Transfer]:
         """
         This is the public interface to construct the final transfer file based on
-        raw (unpenalized) results, slippage penalty, redirected rewards and overdrafts.
+        raw (unpenalized) results, positive, negative slippage, rewards and overdrafts.
         It is very important that the native token transfers are processed first,
-        so that and overdraft from slippage can be carried over and deducted from
+        so that any overdraft from slippage can be carried over and deducted from
         the COW rewards.
         """
-        penalty_total = self._process_native_transfers(indexed_slippage, log_saver)
-        self._process_token_transfers(cow_redirects, log_saver)
-        log_saver.print(
+        penalty_total = self._process_native_transfers(
+            indexed_slippage=index_by(slippage.negative, "solver_address")
```

Comment:
I think what confused me in my original comment is that for me intuitively `slippage.negative` would be the sum of all transactions where slippage was negative (1ETH in my example).

I wonder if this could be made more clear? E.g. why do we have to properties on slippage, if one of them is always expected to be 0? Would it be possible to just expose the total (which can be a negative number)

Reply:
So there is a structure called SplitSlippages which (upon construction from the query results) partitions the results into those which are positive and those which are negative. The reason for this before was to project entirely on to the negative component and forget about the positives. Since we already have this structure in place it was very easy to start making use of the positive slippages. In some sense this is a preliminary form of data processing so that a proper structure can be handed over to the transfer file generation. This way, anything making use of this structure knows what is where and we don't need a bunch of switches in the main logic. 

I hope this clears things up. Let me know if you think the documentation of this structure is unclear or needs updating.

Reply:
I see. What confused me here is the use of singular in the name (slippage). So `slippage.negative` sounds to me like the negative part of a single solver's slippage. What this really is is something like `slippages.solvers_with_negative_total`. Would it be a lot of work to make this rename as part of this PR?

Reply:
I could make some changes in this PR but I am not super fond of this very long name... how about `slippages.negative` and `slippages.positive` (i.e. with the plural)...

Its also not really "solvers with negative slippage" its a list of 
```py
class SplitSlippages:
    """Basic class to store the output of slippage fetching"""

    negative: list[SolverSlippage]
    positive: list[SolverSlippage]
```

and 

```py
class SolverSlippage:
    """Total amount reimbursed for accounting period"""

    solver_address: Address
    solver_name: str
    # ETH amount (in WEI) to be deducted from Solver reimbursement
    amount_wei: int
```

I do notice that the documentation of these structures is way too minimal. I will try to think of a simplified naming convention for this renaming.

Reply:
> but I am not super fond of this very long name...

Why? Writing it takes a few key strokes longer, ok (I bet your awesome editor autocompletes it though) and it means the world to someone trying to read this code later. 
Sure, you can document things with comments, but a reader then has to first find those and also comments (like "this method modifies self") tend to get forgotten about and outdated very quickly. Having to write this variable forces the developer to think about whether it still does what it's called.

Reply:
Maybe to make this a bit easier I will just use the naming convention you suggest.